app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsViewModel.kt

```diff
@@ -13,10 +14,17 @@ class TransactionDetailsViewModel
     appDispatchers: AppDispatchers
 ) : BaseStateViewModel<TransactionDetailsViewState>(appDispatchers) {
 
-    override fun initialState() = TransactionDetailsViewState(null, true)
+    override fun initialState() = TransactionDetailsViewState(null, ViewAction.Loading(true))
+
+    fun loadDetails(txId: String) {
+        safeLaunch {
+            val txDetails = transactionRepository.getTransactionDetails(txId)
```

Comment:
here error handling is missing -> can be added in next pr. Currently, if something happens during tx details load ui won't be updated and spinner will continue to spin

Reply:
The `BaseStateViewModel` emits errors automatically AFAIK with `safeLaunch`. The problem is that in line https://github.com/gnosis/safe-android/blob/bd5323fd5162ca4db94f61574ef5b3f2a3f51110/app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsFragment.kt#L58 The case `viewAction is ShowError` is not handled at all.